src/Unity.Mathematics/quaternion.cs

```diff
@@ -710,6 +710,50 @@ public static float angle(quaternion q1, quaternion q2)
             return diff + diff;
         }
 
+        /// <summary>
+        /// Extracts the rotation from a matrix that potentially has scale.
```

Comment:
"that potentially has scale."

What do you mean by "potentially"? If it has a scale then I assume this extracts the rotation. However, if the matrix _doesn't_ have a scale, what happens?

I'd rephrase this to something like:

Extracts the rotation from a matrix that has a a scale. If the matrix doesn't have a scale then [thing happens].

Presumably returns null or something?

Reply:
this method is the generic form that can support any type of rotation matrix. With no scale, uniform scale, non uniform scale, non singular rotation matrix. It will always returns a valid rotation matrix.

All other methods to extract rotation can fail silently to some degree depending on the input matrix configuration.

Reply:
Ah okay, in that case, I don't understand why we need to mention that it potentially has scale, if it generally supports all types

Reply:
well ... the thing is how do we communicate to our users that if you have non uniform scale in your rotation matrix you need to use this method, all others one will fail and yield bad result

Reply:
Ah ok, well, I feel like we could probably say just that - it seems rather important! my main feedback is that it's currently worded vaguely and we could be more explicit about the fact that you should use this in specific situations e.g 

"Extracts the rotation from a matrix. This method supports any type of rotation matrix: if the matrix has a non uniform scale you should use this method."

Reply:
@jo-unity I updated the docs to use the text you provided above. I split them apart into the summary and remarks sections, hopefully that's ok?